pkg/api/v1beta3/dynakube/activegate/props.go

```diff
@@ -92,20 +101,37 @@ func (ag *Spec) IsMetricsIngestEnabled() bool {
 	return ag.IsMode(MetricsIngestCapability.DisplayName)
 }
 
+func (ag *Spec) IsAutomaticTlsSecretEnabled() bool {
+	return ag.TlsSecretName == "" && ag.trustedCAs != "" && ag.automaticTLSCertificateEnabled
+}
+
 func (ag *Spec) HasCaCert() bool {
-	return ag.IsEnabled() && ag.TlsSecretName != ""
+	return ag.IsEnabled() && (ag.TlsSecretName != "" || ag.IsAutomaticTlsSecretEnabled())
 }
 
-// ActivegateTenantSecret returns the name of the secret containing tenant UUID, token and communication endpoints for ActiveGate.
+// GetTenantSecretName returns the name of the secret containing tenant UUID, token and communication endpoints for ActiveGate.
 func (ag *Spec) GetTenantSecretName() string {
 	return ag.name + TenantSecretSuffix
 }
 
-// ActiveGateAuthTokenSecret returns the name of the secret containing the ActiveGateAuthToken, which is mounted to the AGs.
+// GetAuthTokenSecretName returns the name of the secret containing the ActiveGateAuthToken, which is mounted to the AGs.
 func (ag *Spec) GetAuthTokenSecretName() string {
 	return ag.name + AuthTokenSecretSuffix
 }
 
+// GetTlsSecretName returns the name of the AG TLS secret.
+func (ag *Spec) GetTlsSecretName() string {
```

Comment:
```suggestion
func (ag *Spec) GetTLSSecretName() string {
```
i think we decided in past to use TLS instead of Tls.

Reply:
renamed